### PR TITLE
Hires output

### PR DESCRIPTION
**Pull request by heikoklein**
Coming projects required output in very high resolution, up to 100m. Lagrangian models carry model particles with exact coordinates and are have therefore not restricted to the grid of the meteorological driver. While it is already possible to interpolate the meteorology to high resolution using the FimexIO, this becomes very cost inefficient for such high resolutions. This PR allows therefore to just increase the output resolution to i.e. 25 times the input resolution, while working internally with the original meteorological resolution.

### CODE REVIEW

File: src/common/snapdimML.f90 [hunk 1]
```diff
@@ -17,34 +17,90 @@
...
-    integer, parameter, public :: mavail = 8192
+  integer, parameter, public :: mavail = 8192
+
+!> translate a field in normal resolution to high output resolution
+  public :: hres_field
+
+  contains
+
+!> translate a field in normal resolution to high output resolution
+  subroutine hres_field(field, field_hres, bilinear)
```
Comment (magnusuMET) [inline src/common/snapdimML.f90:59]: `upsample_field`

File: src/common/snap.F90 [hunk 2]
```diff
@@ -128,6 +128,8 @@
 ! FIELD_TIME.FORECAST
 ! FIELD_TIME.VALID
 ! FIELD.OUTPUT= snap.felt
+! * increase output-resolution to factor*input_resolution
+! FIELD.OUTPUT_RESOLUTION_FACTOR
```
Comment (magnusuMET) [inline src/common/snap.F90:132]:
```fortran
! FIELD.OUTPUT_RESOLUTION_FACTOR = 1
```

**Review by magnusuMET — COMMENTED** [review]
Seems like a very useful feature!

It would be easier to keep track of the size of each field if we label them by input/output and drop `_hr`. Introduce `nx_in`, `nx_out` and remove `nx` and references to high res except in snap.input parameter.

Deposition fields are not scaled appropriately

File: src/common/snap.F90 (hunk 2 above)
Comment (heikoklein) [reply src/common/snap.F90:132]: done

File: src/common/snapdimML.f90 (hunk 1 above)
Comment (heikoklein) [reply src/common/snapdimML.f90:62]: I like hres_field better.

**Comment by heikoklein** [comment]
Fixed depositions now.

Refactoring the arrays or introducing a nx_out to make sure which are input and which are output are out of scope of this PR.

**Comment by heikoklein** [comment]
Screenshot from 2023-02-24 15-21-55
Results of SNAP with 100m output grid resolution, i.e. FIELD.OUTPUT_RESOLUTION_FACTOR= 25
Screenshot from 2023-02-24 15-22-21
Results of SNAP with standard 2.5km output grid resolution, i.e. FIELD.OUTPUT_RESOLUTION_FACTOR= 1

Hardly any runtime difference, memory consumption (inkluding 3d-fields) increased from 1G to 3.4G for high resolution.

**Comment by heikoklein** [comment]
Output writing is of course considerably slower in very high resolution, about 1min for this 5min run.

**Pull request merged**